### PR TITLE
wgengine/magicsock: allow disco communication without known endpoints

### PR DESCRIPTION
**Pull request by raggi**
Just because we don't have known endpoints for a peer does not mean that the peer should become unreachable. If we know the peers key, it should be able to call us, then we can talk back via whatever path it called us on. First step - don't drop the packet in this context.

Updates #self

### CODE REVIEW

**Comment by bradfitz** [comment]
What's self? This testable since I don't know the issue.

**Review by maisem — COMMENTED** [review]
LGTM, but really have no clue on the history of this. will defer to @bradfitz or @danderson

**Comment by raggi** [comment]
self == issue is defined in the commit message here. I discovered this on the way to another issue.

I do think it's testable, and we can add tests before it's landed

**Review by danderson — APPROVED** [review]
Change makes sense to me. What likely happened here is that we used to propagate all endpoints eagerly via the control plane, so the test as written was adequate. Then we moved most of the endpoint exchange to be opportunistic at the time of tunnel establishment, meaning valid peers can now be idle with no known endpoints.

**Comment by bradfitz** [comment]
> self == issue is defined in the commit message 

There are many reasons for issues besides containing the text of a commit message.

**Comment by bradfitz** [comment]
But yeah, change itself seems fine. I just want this linked with an issue (story, docs, links, log, metadata, labels) of whatever you're working on. It can be in corp if the details can't be public.

**Pull request merged**
